### ISSUE TITLE
Relative Location header in redirect response causes "Failed to parse URL" false positive

### ISSUE THREAD

**Issue by pddg** [textlint-rule/textlint-rule-no-dead-link#181]
## Bug

When a URL returns a redirect (e.g. 301) with a relative `Location` header (e.g. `Location: /docs/new-path`), the rule reports it as a dead link with the message `Failed to parse URL`.

## Example

```
❯ curl -is https://kubernetes.io/docs/reference/kubernetes-api/policy-resources/network-policy-v1/ | grep -E '301|location'
HTTP/2 301 
location: /docs/reference/kubernetes-api/networking/network-policy-v1/
```

## Root Cause

In `createCheckAliveURL`, the redirect location is resolved via `resolvePath(location, origin)`. When the `Location` header is a root-relative path like `/docs/...`, `isRelative()` returns `false` because Node.js's `path.isAbsolute("/docs/...")` treats it as an absolute file path. As a result, `resolvePath` returns the raw `/docs/...` path unchanged, and the subsequent `fetch("/docs/...")` call throws `TypeError: Failed to parse URL`.

## Impact

Any URL that redirects to a root-relative path (starting with `/`) is incorrectly reported as a dead link, despite being a valid redirect per RFC 7231.

## Fix

Replace `resolvePath(location, base)` with `URL.parse(location, uri)?.href` in the redirect handling logic. The `URL` API correctly resolves both absolute and relative `Location` headers against the original request URI.
